### ISSUE TITLE
Import used by type annotation inside __new__ is considered unused

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#565]
In GitLab by @thauk-copperleaf on Mar 3, 2017, 09:35

*Please describe how you installed Flake8*

Via pip

*Please provide the exact, unmodified output of `flake8 --bug-report`*

```
{
  "dependencies": [
    {
      "dependency": "setuptools",
      "version": "34.1.0"
    }
  ],
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.6.0",
    "system": "Windows"
  },
  "plugins": [
    {
      "plugin": "mccabe",
      "version": "0.6.1"
    },
    {
      "plugin": "pycodestyle",
      "version": "2.3.1"
    },
    {
      "plugin": "pyflakes",
      "version": "1.5.0"
    }
  ],
  "version": "3.3.0"
}
```

*Please describe the problem or feature*

An import used by a type annotation inside `__new__()` in a metaclass is considered unused by flake8.

*If this is a bug report, please explain with examples (and example code) what you expected to happen and what actually happened.*

Minimal test case:

```
from django.db.models import Model


class SomeMetaclass(type):
    def __new__(mcs, name, bases, attrs):
        model_class_attrs: [(str, Model,)] = []
        model_class_attrs.clear()
        return super().__new__(mcs, name, bases, attrs)
```

`example.py:1:1: F401 'django.db.models.Model' imported but unused`

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Mar 4, 2017, 05:59

This is not a Flake8 bug.

**Comment by asottile** [comment]
In GitLab by @thauk-copperleaf on Mar 6, 2017, 10:30

@sigmavirus24 Where can I report the bug, then?

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Mar 6, 2017, 10:52

To the pyflakes project which generates all F* messages?

**Comment by asottile** [comment]
In GitLab by @thauk-copperleaf on Mar 6, 2017, 11:10

In case anyone else is wondering, pyflakes bugs can be reported at: https://bugs.launchpad.net/pyflakes

**Comment by asottile** [comment]
In GitLab by @sjaensch on Oct 20, 2017, 02:05

This bug seems to be fixed in pyflakes 1.6.0, however flake8 restricts pyflakes to a version < 1.6.0. So it really is flake8 now that prevents this issue from being resolved. Could we get a release that works with pyflakes 1.6.0?

**Comment by asottile** [comment]
In GitLab by @asottile on Oct 24, 2018, 07:48

Upstream bugs for pyflakes can be reported to [pycqa/pyflakes](https://github.com/pycqa/pyflakes) on github.

This has been fixed in 3.6.0 by upgrading pyflakes to 2.x

**Comment by asottile** [comment]
In GitLab by @asottile on Oct 24, 2018, 07:48

closed
